### DISCUSSION TITLE
A keymapping on regular mode to move forward/backward by word (key=Left/RightArrow mod="OPT" action= moveForward/Backward by word)

### DISCUSSION THREAD

**Discussion by laueian** [wezterm/wezterm#3724] [Q&A] [answered]
This is a common mapping on every other editor I've used but wezterm appears to only have this keymap/feature in CopyMode, but this is really only useful to me when I am typing a command into the terminal and need to change various parameters.

Example
```zsh
/bin/exe nameOfThing -- param1 param2 etc...
```
When I paste this command into the terminal I start at the end of the line, I would like to jump backward and forward by word without leaving (not certain of the terminology) "insert" mode.

I bet its because you can't use a `CopyMode` command when you are not in CopyMode, but.. can someone shed some light? thanks!

`wezterm.action.CopyMode 'MoveForwardWord'`

```lua
local wezterm = require 'wezterm'

local config = {
    font = wezterm.font 'JetBrains Mono',
    font_size = 25,
    color_scheme = 'WildCherry',
    scrollback_lines = 3500,
    cursor_blink_rate = 800,
    default_cursor_style = 'BlinkingUnderline'
    keys = {
        { key = 'RightArrow', mod = 'OPT', action = wezterm.action.CopyMode 'MoveForwardWord' }
    }
}

return config
```

**Comment by wez** [comment, marked as answer]
Take a look at: https://wezfurlong.org/wezterm/config/lua/keyassignment/SendKey.html

**Reply by laueian** [reply]
Thank you for the blazing fast response.. wow! Worked like a charm.

I wasn't aware of the none `CopyMode` Move Backward/Forward keymapping in wezterm. The only reference I was able to find was
https://wezfurlong.org/wezterm/config/default-keys.html.

Where could I have looked to know that the actions
```
action = act.SendKey {
      key = 'b',
      mods = 'ALT',
    },
action = act.SendKey {
      key = 'f',
      mods = 'ALT',
    },
```
were/are the default wezterm keymappings for Move Forward/Backward?

**Reply by wez** [reply]
Those are not defaults in wezterm. They are defaults in Terminal.app that mac users have become accustomed to using. They are not common defaults on other systems.
